src/test/java/com/microsoft/appcenter/appium/EnhancedAndroidDriverTest.java

```diff
@@ -58,4 +58,28 @@ public void testGetScreenshotAs() throws Exception {
 
         assertTrue("Expected screenshot", reporter.getReports().stream().anyMatch(e -> e.getScreenshotPath() != null && new File(e.getScreenshotPath()).exists()));
     }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testEmptyLabelsNotAllowed() {
+        driver.label("");
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void testNullLabelsNotAllowed() {
+        driver.label(null);
+    }
+
+    @Test
+    public void testVeryLongLabelsAreNotAllowed() {
+        //128 chars
+        String longLabel = "longlonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglonglong";
```

Comment:
Consider this instead: (after the test is moved to EventTest.java :-)

        String longLabel = String.join("", Collections.nCopies(128, "a"));